pkg/distributor/distributor_ingest_storage_test.go

```
@@ -546,6 +546,108 @@ func TestDistributor_Push_ShouldCleanupWriteRequestAfterWritingBothToIngestersAn
 	}
 }
 
+func TestDistributor_Push_IgnoreIngestStorageErrorsDuringMigration(t *testing.T) {
+	t.Parallel()
+
+	ctx := user.InjectOrgID(context.Background(), "user")
+	now := time.Now()
+
+	tests := map[string]struct {
+		shouldFailIngester       bool
+		shouldFailIngestStorage  bool
+		ignoreIngestStorageError bool
+		expectedErrorContext     string
+		maxWaitTime              time.Duration
+	}{
+		"should give precedence to ingester error when both ingester and ingest storage errors occur and IgnoreIngestStorageError is enabled": {
+			shouldFailIngester:       true,
+			shouldFailIngestStorage:  true,
+			ignoreIngestStorageError: true,
+			expectedErrorContext:     "send data to ingesters",
+		},
+		"should succeed when only ingest storage errors occur and IgnoreIngestStorageError is enabled": {
+			shouldFailIngester:       false,
+			shouldFailIngestStorage:  true,
+			ignoreIngestStorageError: true,
+			expectedErrorContext:     "",
+		},
```

One more test, should also succeed if timeout is hit and we're ignoring errors, but there's no classic ingester failure

---

Thanks, added at [63478d3](https://github.com/grafana/mimir/pull/11291/commits/63478d316024afca1df7cb2410c778a9ef16620b)